### PR TITLE
Fix misspelled words in comments, error messages, and test code

### PR DESCRIPTION
**Pull request by seratch**
Fixing typos is sometimes very hard. It's not so easy to visually review them. Recently, I discovered a very useful tool for it, [misspell](https://github.com/client9/misspell). 

This pull request fixes minor typos detected by [misspell](https://github.com/client9/misspell) except for the false positives. If you would like me to work on other files as well, let me know. 

---

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/elasticsearch/blob/master/CONTRIBUTING.md)?
- If submitting code, have you built your formula locally prior to submission with `gradle check`?
- If submitting code, is your pull request against master? Unless there is a good reason otherwise, we prefer pull requests against master and will backport as needed.
- If submitting code, have you checked that your submission is for an [OS that we support](https://www.elastic.co/support/matrix#show_os)?
- If you are submitting this code for a class then read our [policy](https://github.com/elastic/elasticsearch/blob/master/CONTRIBUTING.md#contributing-as-part-of-a-class) for that.

### CODE REVIEW

**Comment by seratch** [comment]
After a few days, this pull request has become having a conflict with other merged changes. Thus, I've rebased on the `master` branch again and pushed it forcibly.

File: server/src/main/java/org/elasticsearch/index/store/Store.java [hunk 1]
```diff
@@ -1063,7 +1063,7 @@ public int size() {
         }
 
         /**
-         * returns the history uuid the store points at, or null if not existant.
+         * returns the history uuid the store points at, or null if not existent.
```
Comment (DaveCTurner) [inline server/src/main/java/org/elasticsearch/index/store/Store.java]: I think `not existent` should be `nonexistent`.

File: server/src/main/java/org/elasticsearch/index/translog/TranslogStats.java [hunk 2]
```diff
@@ -85,12 +85,12 @@ public int estimatedNumberOfOperations() {
         return numberOfOperations;
     }
 
-    /** the size of the generations in the translog that weren't yet to comitted to lucene */
+    /** the size of the generations in the translog that weren't yet to committe to lucene */
```
Comment (DaveCTurner) [inline server/src/main/java/org/elasticsearch/index/translog/TranslogStats.java]: `committed`

File: server/src/main/java/org/elasticsearch/index/translog/TranslogStats.java [hunk 3]
```diff
@@ -85,12 +85,12 @@ public int estimatedNumberOfOperations() {
...
     public long getUncommittedSizeInBytes() {
         return uncommittedSizeInBytes;
     }
 
-    /** the number of operations in generations of the translog that weren't yet to comitted to lucene */
+    /** the number of operations in generations of the translog that weren't yet to committe to lucene */
```
Comment (DaveCTurner) [inline server/src/main/java/org/elasticsearch/index/translog/TranslogStats.java]: `committed`

File: server/src/main/java/org/elasticsearch/rest/action/admin/cluster/RestGetRepositoriesAction.java [hunk 4]
```diff
@@ -51,7 +51,7 @@ public RestGetRepositoriesAction(Settings settings, RestController controller, S
 
     @Override
     public String getName() {
-        return "get_respositories_action";
+        return "get_repositories_action";
```
Comment (DaveCTurner) [inline server/src/main/java/org/elasticsearch/rest/action/admin/cluster/RestGetRepositoriesAction.java]: Although this is a valid correction, this is also a breaking change. I think it only affects the REST Usage API, but I think this should be reverted and fixed separately.

File: server/src/test/java/org/elasticsearch/rest/BaseRestHandlerTests.java [hunk 5]
```diff
@@ -132,7 +132,7 @@ public String getName() {
             hasToString(containsString(
                 "request [/] contains unrecognized parameters: " +
                     "[flied] -> did you mean [field]?, " +
-                    "[respones_param] -> did you mean [response_param]?, " +
+                    "[response_param] -> did you mean [response_param]?, " +
```
Comment (DaveCTurner) [inline server/src/test/java/org/elasticsearch/rest/BaseRestHandlerTests.java]: Lol. This mis-spelling is intentional.

**Review by DaveCTurner — CHANGES_REQUESTED** [review]
In general this looks good. I spotted a handful of changes that are still needed. I'm also not sure whether we should split this up into smaller sets of changes as I worry about merge conflicts and ease of backporting.

**Comment by DaveCTurner** [comment]
Please don't rebase this again, because rebasing loses inline comments from reviews. The preferred way to keep this in line with `master` is to merge master into the PR branch.

**Comment by seratch** [comment]
@DaveCTurner Thank you for your review. Since I have time to work on this now, I will fix the parts you pointed out very soon. Also, I don't rebase this branch anymore.

File: server/src/main/java/org/elasticsearch/index/store/Store.java (hunk 1 above)
Comment (seratch) [reply server/src/main/java/org/elasticsearch/index/store/Store.java]: Thanks. I've fixed it.

File: server/src/main/java/org/elasticsearch/index/translog/TranslogStats.java (hunk 2 above)
Comment (seratch) [reply server/src/main/java/org/elasticsearch/index/translog/TranslogStats.java]: Thanks. I've fixed this and the same one in the same file as well.

File: server/src/main/java/org/elasticsearch/rest/action/admin/cluster/RestGetRepositoriesAction.java (hunk 4 above)
Comment (seratch) [reply server/src/main/java/org/elasticsearch/rest/action/admin/cluster/RestGetRepositoriesAction.java]: Reverted and added a FIXME comment there. If the comment is unnecessary, I will remove it.

File: server/src/test/java/org/elasticsearch/rest/BaseRestHandlerTests.java (hunk 5 above)
Comment (seratch) [reply server/src/test/java/org/elasticsearch/rest/BaseRestHandlerTests.java]: Oops, I reverted it.

**Comment by seratch** [comment]
@DaveCTurner I've pushed a commit which reflects the updates. Could you review them again when you have time? :bow:

File: server/src/main/java/org/elasticsearch/rest/action/admin/cluster/RestGetRepositoriesAction.java [hunk 6]
```diff
@@ -51,7 +51,8 @@ public RestGetRepositoriesAction(Settings settings, RestController controller, S
 
     @Override
     public String getName() {
-        return "get_repositories_action";
+        // FIXIME: the typo, just modifying the response will bring a breaking change to the REST Usage API, see also PR #32792
```
Comment (DaveCTurner) [inline server/src/main/java/org/elasticsearch/rest/action/admin/cluster/RestGetRepositoriesAction.java]: Better not to add this. Also `FIXME` contains a typo ;)

**Comment by DaveCTurner** [comment]
@elasticmachine test this please

File: server/src/main/java/org/elasticsearch/rest/action/admin/cluster/RestGetRepositoriesAction.java (hunk 6 above)
Comment (seratch) [reply server/src/main/java/org/elasticsearch/rest/action/admin/cluster/RestGetRepositoriesAction.java]: Oh... >typo

Thanks, I will remove the line.

Comment (seratch) [reply server/src/main/java/org/elasticsearch/rest/action/admin/cluster/RestGetRepositoriesAction.java]: Removed.

**Comment by DaveCTurner** [comment]
@elasticmachine test this please

**Review by DaveCTurner — CHANGES_REQUESTED** [review]
I did another pass and found a few more things to fix. Additionally I pinged others for second opinions about three changes.

File: modules/reindex/src/test/java/org/elasticsearch/index/reindex/CancelTests.java [hunk 7]
```diff
@@ -122,7 +122,7 @@ private void testCancel(String action, AbstractBulkByScrollRequestBuilder<?, ?>
         logger.debug("waiting for updates to be blocked");
         boolean blocked = awaitBusy(
             () -> ALLOWED_OPERATIONS.hasQueuedThreads() && ALLOWED_OPERATIONS.availablePermits() == 0,
-            1, TimeUnit.MINUTES); // 10 seconds is usually fine but on heavilly loaded machines this can wake a while
```
[the remark this answers is not in the document]
Comment (seratch) [reply modules/reindex/src/test/java/org/elasticsearch/index/reindex/CancelTests.java]: Also fixed.

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/rest/results/RestGetCategoriesAction.java [hunk 8]
```diff
@@ -39,7 +39,7 @@ public RestGetCategoriesAction(Settings settings, RestController controller) {
 
     @Override
     public String getName() {
-        return "xpack_ml_get_catagories_action";
+        return "xpack_ml_get_categories_action";
```
[the remark this answers is not in the document]
Comment (seratch) [reply x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/rest/results/RestGetCategoriesAction.java]: Reverted.

File: x-pack/plugin/src/test/resources/rest-api-spec/test/rollup/start_job.yml [hunk 9]
```diff
@@ -37,7 +37,7 @@ setup:
           }
 
 ---
-"Test start non-existant job":
+"Test start non-existent job":
```
[the remark this answers is not in the document]
Comment (seratch) [reply x-pack/plugin/src/test/resources/rest-api-spec/test/rollup/start_job.yml]: Fixed.

File: x-pack/plugin/src/test/resources/rest-api-spec/test/rollup/stop_job.yml [hunk 10]
```diff
@@ -37,7 +37,7 @@ setup:
           }
 
 ---
-"Test stop non-existant job":
+"Test stop non-existent job":
```
[the remark this answers is not in the document]
Comment (seratch) [reply x-pack/plugin/src/test/resources/rest-api-spec/test/rollup/stop_job.yml]: Fixed.

File: server/src/test/java/org/elasticsearch/rest/BaseRestHandlerTests.java [hunk 11]
```diff
@@ -119,7 +119,7 @@ public String getName() {
         final HashMap<String, String> params = new HashMap<>();
         params.put("consumed", randomAlphaOfLength(8));
         params.put("flied", randomAlphaOfLength(8));
-        params.put("respones_param", randomAlphaOfLength(8));
+        params.put("response_param", randomAlphaOfLength(8));
```
[the remark this answers is not in the document]
Comment (seratch) [reply server/src/test/java/org/elasticsearch/rest/BaseRestHandlerTests.java]: Reverted.

**Comment by DaveCTurner** [comment]
@elasticmachine test this please

**Comment by seratch** [comment]
@DaveCTurner I just resolved the conflict with the latest revision. If there is something I can do for this pull request, please let me know at any time.

**Review by DaveCTurner — APPROVED** [review]
This LGTM, thanks so much for your efforts. I will start a full CI run now.

**Comment by DaveCTurner** [comment]
@elasticmachine test this please

**Comment by DaveCTurner** [comment]
@elasticmachine test this please

**Comment by DaveCTurner** [comment]
`HighlighterSearchIT` fails because it cares too deeply about the lengths of words, and fixing the spelling of `highlighed` to `highlighted` changes this. I reverted this change.

@elasticmachine test this please.

File: x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/ldap/support/LdapMetaDataResolverTests.java [hunk 12]
```diff
@@ -40,7 +40,7 @@ public void testResolveSingleValuedAttributeFromCachedAttributes() throws Except
                 new Attribute("cn", "Clint Barton"),
                 new Attribute("uid", "hawkeye"),
                 new Attribute("email", "clint.barton@shield.gov"),
-                new Attribute("memberOf", "cn=staff,ou=groups,dc=exmaple,dc=com", "cn=admin,ou=groups,dc=exmaple,dc=com")
+                new Attribute("memberOf", "cn=staff,ou=groups,dc=example,dc=com", "cn=admin,ou=groups,dc=example,dc=com")
```
[the remark this answers is not in the document]
Comment (tvernum) [reply x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/ldap/support/LdapMetaDataResolverTests.java]: No, this wasn't deliberate.

**Comment by javanna** [comment]
it seems like this got outdated and there are some merge conflicts. What do you suggest to do @DaveCTurner ? Do you plan on getting back to it?

**Comment by seratch** [comment]
Thank you for your response. I am fine even if you close this PR without merging due to lack of time for it. I understand most of the changes included in this PR are not crucial for the project.

**Comment by DaveCTurner** [comment]
Apologies, I forgot to assign this to myself before going on holiday so it dropped off my list. I will try and bring it in line with master again.

**Comment by DaveCTurner** [comment]
@elasticmachine test this please

**Comment by javanna** [comment]
thanks so much @DaveCTurner !!!

**Comment by seratch** [comment]
@DaveCTurner Thank you for merging this 👍

**Pull request merged**